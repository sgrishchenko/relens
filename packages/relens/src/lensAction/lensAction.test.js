import update, { RELENS_UPDATE } from './lensAction';

describe('lensAction update', () => {
  test(`should create action with the type "${RELENS_UPDATE}"`, () => {
    const value = 'value';
    const actual = update(['some', 'deep', 'prop'])(value);

    expect(actual.type).toBe(RELENS_UPDATE);
  });

  test('should create serializable FLUX action with the path and value specified in the payload', () => {
    const value = 'value';
    const actual = update(['some', 'deep', 'prop'])(value);

    expect(actual).toMatchSnapshot();
  });

  test('should accept the path as the function of the properties', () => {
    const exampleId = 42;
    const value = 'value';
    const actual = update(({ id }) => ['some', 'deep', 'prop', id])(value, { id: exampleId });

    expect(actual).toMatchSnapshot();
  });
});
