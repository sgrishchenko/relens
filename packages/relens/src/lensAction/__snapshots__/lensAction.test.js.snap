// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lensAction update should accept the path as the function of the properties 1`] = `
Object {
  "payload": Object {
    "path": Array [
      "some",
      "deep",
      "prop",
      42,
    ],
    "value": "value",
  },
  "type": "RELENS_UPDATE",
}
`;

exports[`lensAction update should create serializable FLUX action with the path and value specified in the payload 1`] = `
Object {
  "payload": Object {
    "path": Array [
      "some",
      "deep",
      "prop",
    ],
    "value": "value",
  },
  "type": "RELENS_UPDATE",
}
`;
